src/components/controls/oscillator/oscillatorCountControl.tsx

```diff
@@ -0,0 +1,55 @@
+
+import React, { useState, useEffect, useContext } from 'react'
+import { KeyboardContext } from "../../../hooks/keyboardContext";
+import StyledButton from "../../styled/controlButton";
+import IKeyboardContextSignature from "../../../interfaces/IKeyboardContextSignature";
+import StyledLabel from "../../styled/controlLabels";
+import localStorageService from "../../../services/localStorageService";
+
+function OscillatorCountControl(){
+    const localStorageKey = 'numOscillators';
+    const { audioContextWrapper }: IKeyboardContextSignature = useContext(KeyboardContext);
+    let initialOscCount = localStorageService.getNumberByKey(localStorageKey) ?? 1;
+    const [numOscillators, setNumOscillators] = useState(initialOscCount);
+    const minNumOscillators = 1; 
+    const maxNumOscillators = 2; 
+
+    useEffect(() : void=> {
+        audioContextWrapper.setNumOscillators(numOscillators);
```

Comment:
I sent this over text too, but I feel that the usage of of context, state, and storage is a little confusing. Can any of it be condensed? Or moved into a common file to set everything together?

Reply:
Responded over text, but no harm in too my documentation. useState keeps track of state within a component. useContext gives the components access to the global audioContextParameter. useEffect calls the audioContextWrapper method to set the value every time the useState value changes.